python-avd/pyavd/_eos_designs/structured_config/network_services/virtual_source_nat_vrfs.py

```diff
@@ -31,11 +31,12 @@ def virtual_source_nat_vrfs(self: AvdStructuredConfigNetworkServicesProtocol) ->
         if not (self.shared_utils.overlay_vtep and self.shared_utils.network_services_l2 and self.shared_utils.network_services_l3):
             return
 
-        if (loopback_interfaces := self.loopback_interfaces) is None:
+        self.loopback_interfaces()
+        if (loopback_interfaces := self.structured_config.loopback_interfaces) is None:
```

Comment:
This is really bad, since it will run all the logic again. I think we should rather create a small helper function here like 
```python
def _set_virtual_source_nat_for_vrf_loopback(self: whatever, loopback_interface: EosCli.blahblah) -> None:
```
And then call that when we create each loopback interfaces.

Reply:
Fixed!